src/common/entryPoint.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {TelemetryHelper} from "../common/telemetryHelper";
+import {Telemetry} from "../common/telemetry";
+import {Log} from "../common/Log";
+import {OutputChannel} from "vscode";
+
+/* This class should we used for each entry point of the code, so we handle telemetry and error reporting properly */
+export class EntryPoint {
+    private outputChannel: OutputChannel;
+
+    constructor(outputChannel?: OutputChannel) {
+        this.outputChannel = outputChannel;
+    }
+
+    /* This method should wrap any async entry points to the code, so we handle telemetry and error reporting properly */
+    public runCode(taskName: string, errorDescription: string, codeToRun: () => Q.Promise<void>, areErrorsFatal: boolean): void {
+        return this.handleErrors(errorDescription, TelemetryHelper.generate(taskName, codeToRun), /*areErrorsFatal*/ areErrorsFatal);
+    }
+
+    /* This method should wrap any 100% sync entry points to the code, so we handle telemetry and error reporting properly */
+    public runSyncCode(taskName: string, errorDescription: string, codeToRun: () => void): void {
+        try {
+            TelemetryHelper.sendSimpleEvent(taskName + ".starting"); // We call sendSimpleEvent because generate is async only
```

Comment:
I think that if it is possible, this should still be a `TelemetryHelper.generate` to avoid having too many different kinds of telemetry events